pkg/vminfo/vminfo.go

```diff
@@ -21,22 +23,39 @@ import (
 
 	"github.com/google/syzkaller/pkg/host"
 	"github.com/google/syzkaller/pkg/mgrconfig"
+	"github.com/google/syzkaller/pkg/rpctype"
+	"github.com/google/syzkaller/prog"
 	"github.com/google/syzkaller/sys/targets"
 )
 
 type Checker struct {
 	checker
+	checkContext *checkContext
 }
 
 func New(cfg *mgrconfig.Config) *Checker {
+	var impl checker
 	switch {
 	case cfg.TargetOS == targets.Linux:
-		return &Checker{new(linux)}
+		impl = new(linux)
 	default:
-		return &Checker{new(stub)}
+		impl = new(stub)
+	}
+	return &Checker{
+		checker:      impl,
+		checkContext: newCheckContext(cfg, impl),
 	}
 }
 
+func (checker *Checker) RequiredThings() ([]string, []string, []rpctype.ExecutionRequest) {
```

Comment:
Do we really need to distinguish between `infoFiles` and `checkFiles` here? In both non-test usages (`rpc.go` and runtest) one slice is just appended to another.

Reply:
It's appended only if checking is not done yet, otherwise we use just the first slice (info).